### PR TITLE
API for NicConfigurationTemplate CRD

### PR DESCRIPTION
**Pull request by almaslennikov**

### CODE REVIEW

**Comment by almaslennikov** [comment]
@adrianchiris @ykulazhenkov PTAL

File: .gitignore [hunk 1]
```diff
@@ -20,3 +20,6 @@
 # Go workspace file
 go.work
 go.work.sum
+
+# IDE directories
+.idea/
```
Comment (adrianchiris) [inline .gitignore:25]: nit: newline

File: api/v1alpha1/nicconfigurationtemplate_types.go [hunk 2]
```diff
@@ -20,22 +20,68 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+// NicTypeEnum describes an enum of NIC types
+// +enum
+type NicTypeEnum string
+
+type NicSelectorSpec struct {
```
Comment (adrianchiris) [inline api/v1alpha1/nicconfigurationtemplate_types.go:27]: please add description for each field

File: api/v1alpha1/nicconfigurationtemplate_types.go [hunk 3]
```diff
@@ -20,22 +20,96 @@ import (
...
```
Comment (adrianchiris) [inline api/v1alpha1/nicconfigurationtemplate_types.go:27]: nit: also add docstring for public struct types  :)

**Review by adrianchiris — APPROVED** [review]
left one additional comment, feel free to self merge

**Pull request merged**